tests/assets/test_drivers.py

```diff
@@ -41,3 +43,22 @@ def test_gcs_driver(gcs_assetsmanager):
 @skip_unless("ENABLE_S3_TEST", "True")
 def test_s3_driver(s3_assetsmanager):
     _perform_driver_test(s3_assetsmanager.storage_provider.driver)
+
+
+def test_local_driver_overwrite(working_dir):
+    driver = LocalStorageDriver(working_dir)
+    driver.upload_object(
+        os.path.join(TEST_DIR, "assets", "testdata", "some_data.json"), "a/b/c"
+    )
+    # will remove the a/b/c file
```

Comment:
we also might need to check what is going on on disk, in addition to these comments